test/widgets/new_dm_sheet_test.dart

```diff
@@ -0,0 +1,146 @@
+
+Future<void> setupSheet(WidgetTester tester, {
+  required List<User> users,
+  NavigatorObserver? navigatorObserver,
+}) async {
+  addTearDown(testBinding.reset);
+  await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+  final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+  await store.addUser(eg.selfUser);
+  for (final user in users) {
+    await store.addUser(user);
+  }
+
+  await tester.pumpWidget(TestZulipApp(
+    accountId: eg.selfAccount.id,
+    navigatorObservers: navigatorObserver != null ? [navigatorObserver] : [],
+    child: const HomePage()));
+  await tester.pumpAndSettle();
+
+  await tester.tap(find.byIcon(ZulipIcons.user));
+  await tester.pumpAndSettle();
+
+  final fab = find.ancestor(
+    of: find.text("New DM"),
+    matching: find.byType(InkWell),
+  );
+  await tester.tap(fab);
+  await tester.pumpAndSettle();
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('NewDmSheet', () {
+    testWidgets('shows header with correct buttons', (tester) async {
+      await setupSheet(tester, users: []);
+      check(find.descendant(
+        of: find.byType(NewDmPicker),
+        matching: find.text('New DM')
+      )).findsOne();
+
+      check(find.text('Back')).findsOne();
+      check(find.text('Next')).findsOne();
+      check(find.byIcon(ZulipIcons.chevron_left)).findsOne();
+      check(find.byIcon(ZulipIcons.chevron_right)).findsOne();
+
+      final nextButton = tester.widget<GestureDetector>(
+        find.widgetWithText(GestureDetector, 'Next'));
+      check(nextButton.onTap).isNull();
+    });
+
+    testWidgets('shows filtered users based on search', (tester) async {
+      final users = [
+        eg.user(userId: 1, fullName: 'Alice Anderson'),
+        eg.user(userId: 2, fullName: 'Bob Brown'),
+        eg.user(userId: 3, fullName: 'Alice Carter'),
+      ];
+      await setupSheet(tester, users: users);
+
+      await tester.enterText(find.byType(TextField), 'Alice');
+      await tester.pump();
+      check(find.text('Alice Anderson')).findsOne();
+      check(find.text('Alice Carter')).findsOne();
+      check(find.text('Bob Brown')).findsNothing();
+    });
+
+    testWidgets('can select and unselect users', (tester) async {
+      final user = eg.user(userId: 1, fullName: 'Test User');
+      await setupSheet(tester, users: [user]);
+
+      check(find.byIcon(Icons.circle_outlined)).findsOne();
+      check(find.byIcon(Icons.check_circle)).findsNothing();
+
+      final userTile = find.ancestor(
+        of: find.text(user.fullName),
+        matching: find.byType(InkWell),
+      );
+
+      await tester.tap(userTile);
+      await tester.pump();
+      check(find.byIcon(Icons.check_circle)).findsOne();
+      check(find.byIcon(Icons.circle_outlined)).findsNothing();
+
+      final nextButton = tester.widget<GestureDetector>(
+        find.widgetWithText(GestureDetector, 'Next'));
+      check(nextButton.onTap).isNotNull();
```

Comment:
It might be a good idea to check `onTap` before selecting and after deselecting the user tile.